### DISCUSSION TITLE
Number of columns for widgets?

### DISCUSSION THREAD

**Discussion by milenmk** [filamentphp/filament#13405] [Help] [answered]
### Package

Widgets

### Package Version

v3.2.92

### How can we help you?

I have 5 stats widgets on the users resource list page that I want to show on 1 row.

I have added to the List page this function 

'public function getHeaderWidgetsColumns(): int | array
    {
        return [
            'md' => 4,
            'xl' => 5,
        ];
    }'
    
    but I still have 3 widgets per row. What am I doing wrong?

**Comment by leandrocfe** [comment]
`protected int | string | array $columnSpan = 'full';`
?

**Reply by milenmk** [reply]
No effect and I've tried that. Still 3 on the first row and 2 on the second.

**Comment by milenmk** [comment, marked as answer]
I think I've found the issue. For the stats widgets the number of columns is hardcoded:

```
//vendor/filament/widgets/resources/views/stats-overview-widget.blade.php

<x-filament-widgets::widget class="fi-wi-stats-overview">
    <div
        @if ($pollingInterval = $this->getPollingInterval())
            wire:poll.{{ $pollingInterval }}
        @endif
        @class([
            'fi-wi-stats-overview-stats-ctn grid gap-6',
            'md:grid-cols-1' => $columns === 1,
            'md:grid-cols-2' => $columns === 2,
            'md:grid-cols-3' => $columns === 3,
            'md:grid-cols-2 xl:grid-cols-4' => $columns === 4,
        ])
    >
        @foreach ($this->getCachedStats() as $stat)
            {{ $stat }}
        @endforeach
    </div>
</x-filament-widgets::widget>

```

so the settings are not applied. I'll open an issue for this.

**Comment by titoshadow** [comment]
Sorry, did you already open it? I'm following this also :D

**Reply by milenmk** [reply]
I have made a pull request. Currently waiting for the final review.

https://github.com/filamentphp/filament/pull/13415

**Comment by Kane-R-G** [comment]
I ran across this issue but I was able to solve it by updating the `getColumns` function on the widget, like so:

```
    protected function getColumns(): int
    {
        return 1;
    }
```

**Reply by luiseduardobraschi** [reply]
I don't understand how this would solve the issue.
